homeassistant/components/epion/__init__.py

```diff
@@ -0,0 +1,71 @@
+"""The Epion integration."""
+from __future__ import annotations
+
+import socket
+
+from epion import Epion
+from requests.exceptions import ConnectTimeout, HTTPError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import Throttle
+
+from .const import DATA_API_CLIENT, DOMAIN, LOGGER, REFRESH_INTERVAL
+
+CONFIG_SCHEMA = cv.removed(DOMAIN, raise_if_present=False)
+
+PLATFORMS = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Epion from a config entry."""
+    api = Epion(entry.data[CONF_API_KEY])
+
+    try:
+        response = await hass.async_add_executor_job(api.get_current)
+    except (ConnectTimeout, HTTPError, socket.gaierror) as ex:
+        LOGGER.error("Could not retrieve details from Epion API")
+        raise ConfigEntryNotReady from ex
+
+    if len(response["devices"]) == 0:
+        LOGGER.error("Epion account is not active")
+        return False
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = {
+        DATA_API_CLIENT: EpionBase(hass, api, response)
+    }
```

Comment:
If we only store one thing, no need for a dict
```suggestion
    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = EpionBase(hass, api, response)
```

Reply:
Updated this to only hold the new coordinator, without a dict.